### DISCUSSION TITLE
Vamos parar de falar mal de CSS nas conferências, por favor?

### DISCUSSION THREAD

**Discussion by baltazarparra** [frontendbr/forum#319] [Discussão]
Não é incomum conversas nesse teor, Fica o artigo pra reflexão.

Can we stop bad-mouthing CSS in developer talks, please?
https://www.christianheilmann.com/2016/10/05/can-we-stop-bad-mouthing-css-in-developer-talks-please/

**Comment by bernardodiasc** [comment]
O que percebi que é comumente o CSS é criticado por programadores que não entendem realmente CSS, as vezes são pessoas incríveis em JS ou qualquer outra linguagem mas no CSS apenas à escrevem confundidos pela simplicidade da sintaxe. CSS não é simples nem fácil, é uma linguagem declarativa que resulta em definições visuais super complexas, interativas, animadas, 3D, etc.

**Comment by baltazarparra** [comment]
@bernardodiasc Exato, e esse mimimi acaba desanimando e influenciando várias pessoas negativamente.

I challenge anyone to look at the mind-boggling things Ana Tudor does with CSS and tell me that’s not “real programming” and based on a “stupid language”.
http://codepen.io/thebabydino/pen/MeeqLB

**Comment by wmartins** [comment]
@bernardodiasc e @baltazarparra, li um artigo esses dias (https://eager.io/blog/the-languages-which-almost-were-css/) que vai um pouco de encontro ao que vocês estão falando.

Mas enfim, acho que vale entender as limitações e usar ferramentas que auxiliem em resolver o problema.

Dá pra "gostar" de CSS no fim das contas. :P

**Comment by notnotgabriel** [comment]
CSS é LINDO <3

**Comment by felipefialho** [comment]
Quando falam mal de CSS sinto como se estivessem falando mal de mim.

Sério.

**Comment by baltazarparra** [comment]
@wmartins Ótimo artigo, já havia lido, é bem interessante entender a história por trás.

**Comment by baltazarparra** [comment]
@LFeh essa foi a motivação por trás da issue kkkkkkk

**Comment by bernardodiasc** [comment]
Pouco tempo atrás caí num debate de como a arquitetura do CSS é super importante para um projeto React. No ponto de vista de alguém que vem do back-end ou especializado somente em programar JS, por exemplo, não é tão óbvio. Fato é que ao escolher um CSS existente (ex. Bootstrap FW) ou mesmo criar uma estrutura nova pro projeto, precisa-se reconhecer que o CSS e o HTML funcionam junto, a estrutura de um afeta o outro e vice-versa.

Seja React ou PHP, ambos no acabam resultando em HTML que por sua vez utiliza o CSS. Ai vem um PHP meia-boca que resulta em HTML zuado, um CSS importado que não tem nada a ver com o que ta sendo de fato feito (há quem diga que não é possível fazer projetos sem CSS FW... ja ouvi, pode acreditar), acaba o projeto com um monte de hacks e bugs e a famosa imagem do Peter Griffin tentando ajeitar a cortina, e a "culpa é do CSS". Também me revolta haha

**Comment by fititnt** [comment]
> @bernardodiasc commented 12 minutes ago
> O que percebi que é comumente o CSS é criticado por programadores que não entendem realmente CSS (...)

Eu posso listar um monte de tecnologia que isso ocorre :)

Pessoal por exemplo faz piadinha com linguagem Tal, dizendo que é "feia", mas usa ela em sites ou blogs, e quando usa linguagem do coração o máximo que consegue fazer muito bem feito é static generator, porque qualquer coisa além disso ou depende de CMS mal acabado, ou tem que gastar muitas horas pra fazer algo que um novato sobrinho atinge clicando next-next-next.

**Comment by vitor-mendrone-deel** [comment]
Acho que o @afonsopacifer concordaria com 100% desse texto, estou certo?

**Comment by Lucasktrindade** [comment]
Muito das pessoas que falam mal do CSS , é porque não entende totalmente o que essas ferramentas podem fazer, geralmente são pessoas que não estão ligadas ao front. A galera mal sabe que o CSS da para ser usado para fazer interfaces além da web.

**Comment by baltazarparra** [comment]
Mas tem pessoas ligadas ao front também @Lucasktrindade 
Cansei de ver nego usando JavaScript, manipulando o DOM sem necessidade, pra fazer coisas que se resolveriam de boas com HTML/CSS.

**Comment by afonsopacifer** [comment]
@Mendrone SIMMM <3

mas, tenho algumas considerações para jogar no ar tbm rsrsrsr

**O pessoal de Back acha CSS zuado pq:**
- CSS é fácil de aprender
- CSS é complexo de entender
- CSS não faz sentido para um programador
- CSS é client side, logo tem treta de Cross Browser/Device/OS
- Não tem outra opção, se não usar CSS vc chora
- A maioria da galera não acha graça no CSS pq é algo visual e ninguém é designer
- etc...

**O pessoal de Front acha CSS ruim (sem nem se ligar) pq:**

Quase todo mundo acha a API do CSS uma bosta mesmo, por isso todo mundo usa preprocessors. (sem hipocrisia obigos <3)

**Reflexão:**

Aqui nessa issue todos somos **apaixonados por CSS**, legal, mas que tal se nós nos questionarmos sobre usar **CSS de verdade**, não essas coisas de Sass, Stylus, etc..

**Pq usamos preprocessors?**
**R:** Pq é **melhor** que CSS puro

Então vc acha SIM que **CSS é uma bosta** rsrsr

Só para constar, CSS puro é bem melhor do que qualquer pré-processador:
- **calc()** é melhor do que as operações dos preprocessors, pq ele efetua operações entre diferentes unidades e os preprocessors não.
- **@import** é melhor do que os includes da vida pq HTTP2 bla bla, e pq pode combinar com @media para baixar CSS sob demanda. 
- **Custom Variables** são melhoresdo que variáveis compiladas pq são executadas em "run time" e pode ser modificadas com um @media e alterar valores de um calc() por exemplo.

Ps: Não dá ara fazer mixin.. blz, builda só isso com Post CSS sei la..

enfim..
CSS não é um mar de flores.. essas coisas são **complicadas, esquisitas e problemáticas**. 
A API do CSS faz todo o sentido, mas é muito complexa e com toda a certeza do mundo não é perfeita. (se fosse seria intuitivo, o q definitivamente não é)

Moral da história? CSS é amor mas podia ser melhor... talvez com uma API melhor não tivesse a galera do "tentativa e erro" como tem por ai.

Eu sempre falo bem de CSS por ai, mas quando é para falar mau desço o pau tbm rsrsr

ps: Eu amo CSS, mas nem por isso vou passar a mão na cabeça sempre ;)

**Comment by akfzambrana** [comment]
Aprender CSS é facil, tão facil que em dois dias vc aprende, agora, utilizar o CSS não tem nada de facil.

O que o @afonsopacifer disse é bem o que eu penso, eu amo CSS puro, mas utilizo preprocessador não pq é melhor, mas pq no fim das contas facilita/agiliza o trabalho de desenvolvimento, o dia que o CSS tiver o que preciso de um preprocessador então eu vou parar de usar ele (e não adianta falar q ja da pra usar pq onde eu trabalho damos suporte para IE9+ 😢 ).

**Comment by renatogalvones** [comment]
Concordo com tudo, @afonsopacifer!

Talvez um dos pontos que mais me incomodem no CSS seja isso aqui: 

```
CSS
.vader{
    overflow: hidden;
}
.luke{
    float: left;
    width: 25px;
    heigth: 25px;
}

HTML
<div class="vader">
    <div class="luke"></div>
</div>
```

Vc usa o **overflow:hidden** no elemento pai para que ele respeite a altura do filho. Legal.

Maaaaaas, traduzindo a propriedade ao pé da letra seria algo como:

> "tudo que tentar vazar do elemento pai, vc ESCONDE, blz?"

Só que na real vc tá usando uma porra de um HIDDEN para MOSTRAR coisas... hauhauhauahuahuahuahua

Claro, dá pra resolver de outras formas(ex: ::after com clear:both ), mas já foi boa prática fazer dessa forma. =P

Mas no final das contas tem uns desafios legais de resolver. Eu curto. =]

:metal:

**Comment by felipefialho** [comment]
@afonsopacifer  Lembrei de uma talk do @bernarddeluna no 7Masters 😹

**Comment by akfzambrana** [comment]
@renatogalvones no caso do seu exemplo, a culpa não é 100% do CSS e sim boa parte, quase 100%, do uso "errado" q nós fazemos da propriedade `float`. 😉

**Comment by renatogalvones** [comment]
@akfzambrana Sem dúvida. A treta é que já conheci muita gente que foi **ensinada** dessa maneira. Até em faculdade. E nem ensinavam as alternativas. =/

Acredito que hoje em dia os conteúdos já devem ter sido atualizados(espero) mas vai saber né!? Uma galera deve chorar sangue até hoje.

**Comment by vitor-mendrone-deel** [comment]
Fazer piadas com CSS é semelhante a reclamar de tal "JS Fatigue". Minha opinião particular sobre isso é que só reclama quem não entende ambos ou não alcançou maturidade o suficiente pra decidir os caminhos a seguir. 

Mas acho as reclamações e piadas extremamente necessárias pois acabam expondo problemas, dificuldades e barreiras que devem ser transpostas pelo CSSWG. 

Essas piadas evitam (ou ajudam a evitar) que [ESSA LISTA](https://wiki.csswg.org/ideas/mistakes) continue crescendo e que gifs como o abaixo causem identificação instantânea em todos nós: 

![css](https://cloud.githubusercontent.com/assets/4412692/19154290/48635396-8bb0-11e6-8a2d-c78b3dd1c98a.gif)

**Comment by gabrielmanara** [comment]
@renatogalvones estou cursando Sistemas de Informação em uma faculdade particular, sinto lhe informar que os conteúdos não foram atualizados. É trágico, tem faculdades que mal ensinam js.
Posso dizer que muitas vezes esse "preconceito" com CSS vem de dentro dos cursos, que são totalmente voltados para back-end e muitos professores tem certa desdenha com o CSS (pobre CSS). Os cursos de TI hoje estão parados no tempo em questão de front-end.

**Comment by renatogalvones** [comment]
@gabrielmanara Triste isso, mano. Ainda bem que a comunidade Front-end é uma das coisas mais bacanas que existem e de certa forma, supri esses pontos que as instituições deixam passar.

De verdade, sem demagogia, isso aqui(comunidade) é o que mais me motiva a estudar, ir em eventos, até palestrar e organizar eventos(Rolou ontem o 1º Meetup Alphaville :metal: ).

A galera, em termos gerais, é colaborativa pra caralho e tem sempre alguns puxando os outros a aprenderem mais. Acho isso foda demais!

**Comment by suissa** [comment]
CSS é a coisa mais simples de entender e a mais difícil de dominar e olhe que eu venho dos Hacks do IE6 e sabia todos de cabeça hehehehhee.

Pra quem não curte CSS é só usar PostCSS. ;)

**Comment by ingridrauany** [comment]
Essa coisa que o @gabrielmanara disse sobre a faculdade é totalmente real, também faço Sistemas de Informação, tive dois períodos de programação web e não aprendi nada de front-end. Então estou procurando estudar por conta própria e percebi que na faculdade os professores parecem ou fingem parar no tempo.

A minha maior dificuldade nesse início é querer usar propriedades do CSS apenas "chutando" e não entendendo que acontece e como ao definir algo você interfere nos demais elementos. Se não tiver esforço para entender os comportamentos dos elementos o CSS realmente não vai ajudar (e nem tem como).

**Comment by vitor-mendrone-deel** [comment]
E esse é realmente o caminho. CSS se aprende no mundo real, fazendo coisas reais, é o tipo de coisa que não se aprende em livros/apostilas/slides

Um conselho que sempre dou pra galera que está na faculdade: Estudem tudo por conta própria. Faculdades de TI no BR são uma enganação.
